src/protocol_auth.c

```diff
@@ -143,56 +144,35 @@ static bool send_proxyrequest(meshlink_handle_t *mesh, connection_t *c) {
 }
 
 bool send_id(meshlink_handle_t *mesh, connection_t *c) {
-
-	int minor = mesh->self->connection->protocol_minor;
-
 	if(mesh->proxytype && c->outgoing)
 		if(!send_proxyrequest(mesh, c)) {
 			return false;
 		}
 
-	return send_request(mesh, c, NULL, "%d %s %d.%d %s", ID, mesh->self->connection->name, mesh->self->connection->protocol_major, minor, mesh->appname);
+	return send_request(mesh, c, NULL, "%d %s %d.%d %s", ID, mesh->self->name, PROT_MAJOR, PROT_MINOR, mesh->appname);
 }
```

Comment:
In meshlink_join, the minor number is hard coded to 1 where as here we are using PROT_MINOR. Shouldn't both requests contain same protocol major and minor number.